File: TestFlutter/lib/beans/FileBean.dart
import 'dart:convert';

class FileBean {
  String name; // 文件名
  String absolutePath; // 文件绝对路径
  String suffix; // 文件后缀名
  bool isDirectory; // 是否是文件夹

  int fileLength; // 文件大小
  int lastModified; // 最近修改的时间

  FileBean(this.name, this.absolutePath, this.suffix, this.isDirectory,
      this.fileLength, this.lastModified);

  @override
  String toString() {
    return 'FileBean{name: $name, absolutePath: $absolutePath, suffix: $suffix, isDirectory: $isDirectory, fileLength: $fileLength, lastModified: $lastModified}';
  }

  static FileBean fromJson(Map<String, dynamic> map) {
    return FileBean(
        map["name"], map["absolutePath"], map["suffix"], map["isDirectory"],map["fileLength"],map["lastModified"]);
  }

  static List<FileBean> fromJsonArr(List<dynamic> list) {
    List<FileBean> newList = [];
    for (var map in list) {
      newList.add(fromJson(map));
    }
    return newList;
  }
}

File: NetworkDiskClientAndroid/app/src/main/java/com/hld/networkdisk/client/beans/FileBean.kt
package com.hld.networkdisk.client.beans

data class FileBean(
    val name: String, // 文件名
    val absolutePath: String, // 文件绝对路径
    val suffix: String, // 文件后缀名
    val isDirectory: Boolean, // 是否是文件夹
    val fileLength: Long, // 文件大小
    val lastModified: Long, // 最近修改的时间
    var previewImageBase64: String? = null // 预览的base64
)